usr/src/uts/common/fs/zfs/metaslab.c

```diff
@@ -199,6 +200,43 @@ static void metaslab_set_fragmentation(metaslab_t *);
 kmem_cache_t *metaslab_alloc_trace_cache;
 
 /*
+ * How many TXG's worth of updates should be aggregated per TRIM/UNMAP
+ * issued to the underlying vdev. We keep two range trees of extents
+ * (called "trim sets") to be trimmed per metaslab, the `current' and
+ * the `previous' TS. New free's are added to the current TS. Then,
+ * once `zfs_txgs_per_trim' transactions have elapsed, the `current'
+ * TS becomes the `previous' TS and a new, blank TS is created to be
+ * the new `current', which will then start accumulating any new frees.
+ * Once another zfs_txgs_per_trim TXGs have passed, the previous TS's
+ * extents are trimmed, the TS is destroyed and the current TS again
+ * becomes the previous TS.
+ * This serves to fulfill two functions: aggregate many small frees
+ * into fewer larger trim operations (which should help with devices
+ * which do not take so kindly to them) and to allow for disaster
+ * recovery (extents won't get trimmed immediately, but instead only
+ * after passing this rather long timeout, thus preserving
+ * 'zfs import -F' functionality).
+ */
+unsigned int zfs_txgs_per_trim = 32;
```

Comment:
Yes, this is exactly what it's based on. The value is a tradeoff between:
1) Keeping the trim commands small (this would imply fewer txgs per trim)
2) Keeping the ability to rollback via zpool import -F for as many txgs as possible (this would imply more txgs per trim).
3) Waiting around too long that the user starts to get uneasy about not seeing any space being freed after they remove some files.
Ultimately, 32 seems to produce a reasonable trim command size, reacts within a couple of minutes of removing data and doesn't impact uberblock rollback on pretty much any SSD out there (so far all SSDs I've ever seen were sectorsize=4k or more).